File: flutter17_mobile/lib/models/laptop.dart
import 'package:json_annotation/json_annotation.dart';

part 'laptop.g.dart';

@JsonSerializable()
class Laptop {
  int? id;
  int? productId;

  // Performance
  String? processor;
  int? ram;
  String? storageType;
  int? storageCapacity;
  String? graphicsCard;

  // Display
  String? screenSize;
  String? screenResolution;
  String? screenType;

  // Battery
  int? batteryCapacity;
  int? batteryLife;

  // Connectivity
  bool? hasWiFi;
  bool? hasBluetooth;
  int? usbPorts;
  bool? hasEthernetPort;
  bool? hasHDMI;
  bool? hasThunderbolt;

  // Physical Characteristics
  double? weight;
  String? dimensions;
  String? buildMaterial;

  // Additional Features
  bool? hasBacklitKeyboard;
  bool? hasFingerprintReader;
  bool? hasWebcam;
  String? operatingSystem;

  Laptop({
    this.id,
    this.productId,
    this.processor,
    this.ram,
    this.storageType,
    this.storageCapacity,
    this.graphicsCard,
    this.screenSize,
    this.screenResolution,
    this.screenType,
    this.batteryCapacity,
    this.batteryLife,
    this.hasWiFi,
    this.hasBluetooth,
    this.usbPorts,
    this.hasEthernetPort,
    this.hasHDMI,
    this.hasThunderbolt,
    this.weight,
    this.dimensions,
    this.buildMaterial,
    this.hasBacklitKeyboard,
    this.hasFingerprintReader,
    this.hasWebcam,
    this.operatingSystem,
  });

  factory Laptop.fromJson(Map<String, dynamic> json) => _$LaptopFromJson(json);

  Map<String, dynamic> toJson() => _$LaptopToJson(this);
}

File: angular17_web/src/app/models/laptop/laptop.ts
export class Laptop {
  id!: number;
  productId!: number;
  processor!: string;
  ram!: number;
  storageType!: string;
  storageCapacity!: number;
  graphicsCard!: string;
  screenSize!: string;
  screenResolution!: string;
  screenType!: string;
  batteryCapacity!: number;
  batteryLife!: number;
  hasWiFi!: boolean;
  hasBluetooth!: boolean;
  usbPorts!: number;
  hasEthernetPort!: boolean;
  hasHDMI!: boolean;
  hasThunderbolt!: boolean;
  weight!: number;
  dimensions!: string;
  buildMaterial!: string;
  hasBacklitKeyboard!: boolean;
  hasFingerprintReader!: boolean;
  hasWebcam!: boolean;
  operatingSystem!: string;

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.productId = data.productId;
    this.processor = data.processor;
    this.ram = data.ram;
    this.storageType = data.storageType;
    this.storageCapacity = data.storageCapacity;
    this.graphicsCard = data.graphicsCard;
    this.screenSize = data.screenSize;
    this.screenResolution = data.screenResolution;
    this.screenType = data.screenType;
    this.batteryCapacity = data.batteryCapacity;
    this.batteryLife = data.batteryLife;
    this.hasWiFi = data.hasWiFi;
    this.hasBluetooth = data.hasBluetooth;
    this.usbPorts = data.usbPorts;
    this.hasEthernetPort = data.hasEthernetPort;
    this.hasHDMI = data.hasHDMI;
    this.hasThunderbolt = data.hasThunderbolt;
    this.weight = data.weight;
    this.dimensions = data.dimensions;
    this.buildMaterial = data.buildMaterial;
    this.hasBacklitKeyboard = data.hasBacklitKeyboard;
    this.hasFingerprintReader = data.hasFingerprintReader;
    this.hasWebcam = data.hasWebcam;
    this.operatingSystem = data.operatingSystem;
  }
}
